### ISSUE TITLE
Fails to Install in Proxmox 8.1, Kernel 6.5.13-1-pve: Too Few Arguments to ‘drm_dp_calc_pbn_mode’ when called by  ‘intel_dp_mst_find_vcpi_slots_for_bpp’

### ISSUE THREAD

**Issue by johntdavis84** [strongtz/i915-sriov-dkms#155]
I got as far as trying to install and build the new kernel, and ended up failing with a build error. Unfortunately, I accidentally cleared it before I could copy it.

However, I wanted to check and confirm whether anyone actually has this working with my configuration before I try again.

The hardware is an Intel i5-12500T (UHD 770), but as I didn't make it past installing the DKMS module, I don't think that's relevant to the issue. I hope.

**EDIT 1 @ Monday, April 22, 2024**
I've got a trace from today's build failure. The latest merge changed something. Now I'm seeing a "too few arguments" error in the "drm_dp_calc_pbn_mode" function.

This sounds like something that might be easy to fix?

```
  CC [M]  /var/lib/dkms/i915-sriov-dkms/6.5.13-1/build/drivers/gpu/drm/i915/display/intel_dp_mst.o
/var/lib/dkms/i915-sriov-dkms/6.5.13-1/build/drivers/gpu/drm/i915/display/intel_dp_mst.c: In function ‘intel_dp_mst_find_vcpi_slots_for_bpp’:
/var/lib/dkms/i915-sriov-dkms/6.5.13-1/build/drivers/gpu/drm/i915/display/intel_dp_mst.c:85:31: error: too few arguments to function ‘drm_dp_calc_pbn_mode’
   85 |             crtc_state->pbn = drm_dp_calc_pbn_mode(adjusted_mode->crtc_clock,
      |                               ^~~~~~~~~~~~~~~~~~~~
In file included from /var/lib/dkms/i915-sriov-dkms/6.5.13-1/build/drivers/gpu/drm/i915/display/intel_display_types.h:36,
                 from /var/lib/dkms/i915-sriov-dkms/6.5.13-1/build/drivers/gpu/drm/i915/display/intel_dp_mst.c:40:
./include/drm/display/drm_dp_mst_helper.h:835:5: note: declared here
  835 | int drm_dp_calc_pbn_mode(int clock, int bpp, bool dsc);
      |     ^~~~~~~~~~~~~~~~~~~~
/var/lib/dkms/i915-sriov-dkms/6.5.13-1/build/drivers/gpu/drm/i915/display/intel_dp_mst.c: In function ‘intel_dp_mst_mode_valid_ctx’:
/var/lib/dkms/i915-sriov-dkms/6.5.13-1/build/drivers/gpu/drm/i915/display/intel_dp_mst.c:898:13: error: too few arguments to function ‘drm_dp_calc_pbn_mode’
  898 |             drm_dp_calc_pbn_mode(mode->clock, min_bpp) > port->full_pbn) {
      |             ^~~~~~~~~~~~~~~~~~~~
./include/drm/display/drm_dp_mst_helper.h:835:5: note: declared here
  835 | int drm_dp_calc_pbn_mode(int clock, int bpp, bool dsc);
      |     ^~~~~~~~~~~~~~~~~~~~
make[2]: *** [scripts/Makefile.build:251: /var/lib/dkms/i915-sriov-dkms/6.5.13-1/build/drivers/gpu/drm/i915/display/intel_dp_mst.o] Error 1
make[1]: *** [/usr/src/linux-headers-6.5.13-1-pve/Makefile:2039: /var/lib/dkms/i915-sriov-dkms/6.5.13-1/build] Error 2
make: *** [Makefile:234: __sub-make] Error 2
```

**EDIT 2:** I found the problematic function call in intel_dp_mst.c . It's only getting two arguments, but needs three.
```
  84   │     for (bpp = max_bpp; bpp >= min_bpp; bpp -= step) {
  85   │         crtc_state->pbn = drm_dp_calc_pbn_mode(adjusted_mode->crtc_clock,
  86   │                                    dsc ? bpp << 4 : bpp);
```

Here's my config:

```
pveversion -v
proxmox-ve: 8.1.0 (running kernel: 6.5.13-1-pve)
pve-manager: 8.1.4 (running version: 8.1.4/ec5affc9e41f1d79)
proxmox-kernel-helper: 8.1.0
proxmox-kernel-6.5.13-1-pve-signed: 6.5.13-1
proxmox-kernel-6.5: 6.5.13-1
proxmox-kernel-6.5.11-8-pve-signed: 6.5.11-8
```

**Comment by ich777** [comment]
Maybe related to #151 and should be fixed with #152 as long as you are on Kernel 6.5.13-5-pve which you are not from what I see from the description.

**Comment by johntdavis84** [comment]
> Maybe related to #151 and should be fixed with #152 as long as you are on Kernel 6.5.13-5-pve which you are not from what I see from the description.

Thanks!

That definitely did something. I'm getting a different (and hopefully easier to fix) error about a function having too few arguments now.

I've updated my original comment with the details.

**Comment by ich777** [comment]
> proxmox-ve: 8.1.0 (running kernel: 6.5.13-1-pve)

Sorry for the maybe dumb question but can't you just update to a version which includes Kernel version Kernel 6.5.13-**5**-pve instead of 6.5.13-**1**-pve to solve your issue.
I linked the PR above because it states:
`Prcoceeding with this merge may break builds in PVE with kernel < 6.5.13-5-pve`

**Comment by johntdavis84** [comment]
> > proxmox-ve: 8.1.0 (running kernel: 6.5.13-1-pve)
> 
> Sorry for the maybe dumb question because I don't use Proxmox but can't you just update to a Proxmox version which includes Kernel version Kernel 6.5.13-**5**-pve instead of 6.5.13-**1**-pve to solve your issue. I linked the PR above because it states: `Prcoceeding with this merge may break builds in PVE with kernel < 6.5.13-5-pve`

...
That is not a dumb question. I'm a bit of an idiot, apparently.

My version is the latest version I can get when i do a `dist-upgrade`, but I'm also not on the test (beta) repo. 
Let me look and see what I can find out. I'll report back.

**Comment by johntdavis84** [comment]
Thanks everyone. I'm going to close this as solved shortly. :)

I was running on the latest enterprise version of the 6.5.13 PVE kernel, which wasn't new enough. (I think it was the "_1" variant.) On Proxmox 8.1, the newer ones were in the test repo, which I keep disabled.

The day Proxmox 8.2 was released, 6.5.13-5-pve became available in the non-test repos (or at least, the enterprise repo). 

So I was able to get it added and installed and the system boots.

I'm going to suggest to the guide author I used to add a section about requiring 6.5.13-5-pve, which is currently not mentioned, with instructions on how to install it if you're on a fresh install of Proxmox 8.2, which doesn't include the 6.5.13-5-pve by default. It can be installed with a single command, though. :)
